sale_import_base/tests/common_sale_order_import.py

```diff
@@ -0,0 +1,107 @@
+from odoo.addons.sale.tests.test_sale_common import TestCommonSaleNoChart
+
+from .data import data
+
+
+class SaleImportCase(
+    SavepointDatamodelCase, TestCommonSaleNoChart, SavepointComponentCase
+):
+    @classmethod
+    def setUpClass(cls):
+        super(SaleImportCase, cls).setUpClass()
+        cls.setUpClassicProducts()
+        cls.setUpTaxes()
+        cls.setUpFpos()
+        cls.setUpPaymentAcquirer()
+        cls.setUpMisc()
+        cls.sale_order_example_vals = data
+        cls.sale_order_example_vals["pricelist_id"] = cls.env.ref("product.list0").id
+
+    @classmethod
+    def setUpTaxes(cls):
+        tax_obj = cls.env["account.tax"]
+        tax_vals = {
+            "name": "Fictional tax 9%",
+            "amount": "9.00",
+            "type_tax_use": "sale",
+            "company_id": cls.env.ref("base.main_company").id,
+        }
+        cls.tax = tax_obj.create(tax_vals)
+        cls.product_order.taxes_id = cls.tax
+
+    @classmethod
+    def setUpFpos(cls):
+        tax_obj = cls.env["account.tax"]
+        fiscal_pos_obj = cls.env["account.fiscal.position"]
+        fiscal_pos_line_obj = cls.env["account.fiscal.position.tax"]
+
+        # CH
+        fpos_vals_swiss = {
+            "name": "Swiss Fiscal Position",
+            "country_id": cls.env.ref("base.ch").id,
+            "zip_from": 0,
+            "zip_to": 0,
+            "auto_apply": True,
+        }
+        cls.fpos_swiss = fiscal_pos_obj.create(fpos_vals_swiss)
+        tax_vals_swiss = {
+            "name": "Swiss Export",
+            "amount": "0.00",
+            "type_tax_use": "sale",
+            "company_id": cls.env.ref("base.main_company").id,
+        }
+        cls.tax_swiss = tax_obj.create(tax_vals_swiss)
+        fpos_line_vals = {
+            "position_id": cls.fpos_swiss.id,
+            "tax_src_id": cls.tax.id,
+            "tax_dest_id": cls.tax_swiss.id,
+        }
+        fiscal_pos_line_obj.create(fpos_line_vals)
+
+    @classmethod
+    def setUpPaymentAcquirer(cls):
+        PaymentAcquirer = cls.env["payment.acquirer"]
+
+        # Acquirer and mode of payment
+        acquirer_vals = {
+            "name": "credit_card",
+            "provider": "manual",
+            "company_id": cls.env.ref("base.main_company").id,
+            "environment": "test",
+            "payment_flow": "s2s",
+        }
+        PaymentAcquirer.create(acquirer_vals)
+
+    @classmethod
+    def setUpMisc(cls):
+        cls.env = cls.env(context=dict(cls.env.context, test_queue_job_no_delay=True))
+        dummy_chunk = cls.env["queue.job.chunk"].create(
+            {
+                "usage": "basic_create",
+                "apply_on_model": "res.partner",
+                "data_str": '{"name": "Dummy Partner"}',
+                "record_id": cls.env.ref("sale_channel.sale_channel_ebay"),
+                "model_name": "sale.channel",
+            }
+        )
+        with dummy_chunk.work_on("sale.order") as work:
```

Comment:
This code is weird. Creating a fake hcunk to call the import after.
It's will be better to
- create a junk (with that data of the sale order)
- process the chunk directly by calling the method "_enqueue_job" without delay.

Also the method "_enqueue_job" here have a bad name : https://github.com/akretion/sale-import/blob/9ac095adb356162fcafd74aee1a8029d4c6ccb49/queue_job_chunk/models/queue_job_chunk.py#L58

We should called it "process_chunk" ? what do you think

Reply:
Also in it will be better to not pass any arg to the processor
https://github.com/akretion/sale-import/blob/9ac095adb356162fcafd74aee1a8029d4c6ccb49/queue_job_chunk/models/queue_job_chunk.py#L65

because in the processor we can to self.collection.data_str to access them

Reply:
I don't have any real opinion on method name, I renamed it as you said

About the fake chunk. I did this to isolate testing the chunk functionality and test only the importer service. I will change it now.